File: JavaScript/8js 1/addLength.js
/*
|--------------------------------------------------------------------------
| Add Length
|--------------------------------------------------------------------------
|
*/

function addLength(str) {
   return str.split(" ").map((i) => `${i} ${i.length}`);
}


function addLength(str) {
   let strArr = str.split(" ");
   let res = [];
   for (let i of strArr) {
      res.push(`${i} ${i.length}`);
   }
   return res;
}


function addLength(str) {
   return str.split(" ").map((v) => v + " " + v.length);
}


console.log(addLength("apple ban"), ["apple 5", "ban 3"]);
console.log(addLength("you will win"), ["you 3", "will 4", "win 3"]);

File: Python/8py 1/add_length.py
"""
|--------------------------------------------------------------------------
| Add Length
|--------------------------------------------------------------------------
|
"""

def add_length(str_):
   return [f"{i} {len(i)}" for i in str_.split(' ')]


def add_length(str_):
   res = []
   for word in str_.split():
      res.append(word + ' ' + str(len(word)))
   return res


def add_length(str_):
   return ['{} {}'.format(word, len(word)) for word in str_.split()]


print(add_length('apple ban'),["apple 5", "ban 3"])
print(add_length('you will win'),["you 3", "will 4", "win 3"])
print(add_length('you'),["you 3"])
print(add_length('y'),["y 1"])